File: is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int i = 0; 

        if  (s.length() > t.length()) {
            return false;
        }
        if(s.length() == 0){
            return true;
        }

        for (int j = 0; j < t.length(); j++) {
            if (s.charAt(i) == t.charAt(j)) {
                i++;
                if (i == s.length()) {
                    return true;
                }
            }
        }
        return false;
    
    }
}

File: 392-is-subsequence/is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {

    let l = 0;
    let r = 0;

    while(l < s.length && r < t.length){
        if(s[l] === t[r]){
            l++;
        }

        r++;
    }

    if(l === s.length){
        return true;
    } else {
        return false;
    }

};